x-pack/agentbeat/Jenkinsfile.yml

```diff
@@ -0,0 +1,33 @@
+when:
+    branches: true             ## for all the branches
+    changeset:                 ## when PR contains any of those entries in the changeset
+        - "^agentbeat/.*"
+        - "@ci"                ## special token regarding the changeset for the ci
+        - "@oss"               ## special token regarding the changeset for the oss
+    comments:                  ## when PR comment contains any of those entries
+        - "/test agentbeat"
+    labels:                    ## when PR labels matches any of those entries
+        - "agentbeat"
+    parameters:                ## when parameter was selected in the UI.
+        - "agentbeat"
+    tags: true                 ## for all the tags
+platform: "immutable && ubuntu-22" ## default label for all the stages
+stages:    
+    packaging-linux:
+        packaging-linux: "mage package"
+        e2e:
+            enabled: false
+        stage: packaging
+        when:
+            branches: false    ## Only on a PR basis for the time being
+            tags: false        ## packaging on branches/tags is already in place with the downstream build.
+            changeset:         ## when PR contains any of those entries in the changeset
+              - "^x-pack/agentbeat/.*"
+              - "^auditbeat/.*"
+              - "^filebeat/.*"
+              - "^heartbeat/.*"
+              - "^metricbeat/.*"
+              - "^osquerybeat/.*"
+              - "^packetbeat/.*"
+              - "@oss"               ## special token regarding the changeset for the oss
+              - "@xpack"    
```

Comment:
The changeset at the stage level should be a subset of the changeset in the top-level `when`. 

Can you add these changeset values in `when.changeset`?

Reply:
Done